cli/cmd/generate.go

```diff
@@ -248,16 +261,67 @@ func patchTargets(paths []string, logger *slog.Logger) error {
 			return fmt.Errorf("failed to unmarshal %s: %w", path, err)
 		}
 
-		var changed bool
-		replaceRuntimeClassName := runtimeClassNamePatcher(&changed)
+		for i, resource := range kubeObjs {
+			deploy, ok := resource.(*applyappsv1.StatefulSetApplyConfiguration)
+			if ok && deploy.Spec.Template.Annotations[contrastRoleAnnotationKey] == "coordinator" {
+				continue
+			}
+			resource, err = kuberesource.AddInitializer(resource, kuberesource.Initializer())
+			if err != nil {
+				return err
+			}
+			kubeObjs[i] = resource
+		}
+
+		logger.Debug("Updating resources in yaml file", "path", path)
+		modifiedData, err := kuberesource.EncodeResources(kubeObjs...)
+		if err != nil {
+			return err
+		}
+		if err := os.WriteFile(path, modifiedData, os.ModePerm); err != nil {
```

Comment:
I think it would be better to only write those files once. Right now, we are writing here with images that don't make sense, and later we read those files again and patch the images. If the command fails in between, the user ends up with partially patched files that contain images that aren't recommended to use.

As discussed out of band, a solution might be to move the `injectInitializer` func into patchTargets.

This would also solve the current naming problem, as it is a bit unclear what `patchTargets` does as we also patch targets in `injectInitializer`.